pkg/controllers/provisioningv2/rke2/machinedrain/machinedrain.go

```diff
@@ -22,44 +22,51 @@ import (
 )
 
 type handler struct {
-	ctx      context.Context
-	machines capicontrollers.MachineClient
-	secrets  corecontrollers.SecretCache
+	ctx          context.Context
+	machineCache capicontrollers.MachineCache
+	secrets      corecontrollers.SecretClient
+	secretCache  corecontrollers.SecretCache
 }
 
 func Register(ctx context.Context, clients *wrangler.Context) {
 	h := &handler{
-		ctx:      ctx,
-		machines: clients.CAPI.Machine(),
-		secrets:  clients.Core.Secret().Cache(),
+		ctx:          ctx,
+		machineCache: clients.CAPI.Machine().Cache(),
+		secrets:      clients.Core.Secret(),
+		secretCache:  clients.Core.Secret().Cache(),
 	}
-	clients.CAPI.Machine().OnChange(ctx, "machine-drain", h.OnChange)
+	clients.Core.Secret().OnChange(ctx, "machine-drain", h.OnChange)
```

Comment:
We may want to generate the core client/secret thing with a selective indexer to keep down on the noise

Reply:
OK. I have a solution for your review.